shared/api/dfx/ApiService.ts

```diff
@@ -76,32 +78,84 @@ export const getUserDetail = async (): Promise<UserDetail> => {
   return await fetchFrom<UserDetailDto>(`${UserUrl}/detail`).then(fromUserDetailDto)
 }
 
-export const putUser = async (user: User): Promise<UserDetail> => {
-  return await fetchFrom<UserDetailDto>(UserUrl, 'PUT', toUserDto(user)).then(fromUserDetailDto)
+export const putUser = async (user: User | UserDetailRequestDto): Promise<UserDetail> => {
+  return await fetchFrom<UserDetailDto>(UserUrl, 'PUT', toUserDto(user as User)).then(fromUserDetailDto)
 }
 
 export const updateRefFee = async (fee: number): Promise<void> => {
   return await fetchFrom(UserUrl, 'PUT', { refFeePercent: fee })
 }
 
 // --- KYC --- //
-export const putKycData = async (data: KycData): Promise<void> => {
+export const putKycData = async (data: KycData, code?: string): Promise<KycInfo> => {
```

Comment:
As discussed, I would remove the legacy calls and switch to required code parameter.

Reply:
As I see no changes in Swagger to the code and non-code version, I kinda **have** to leave this in for now 🤷‍♂️ .

I can resort them, when they have been updated (as we discussed, we wanted to keep some legacy calls, like _post_).

Reply:
Don't really understand the argument, but I'm fine if we resort this later 😉